server/service/src/stocktake/insert/generate.rs

```diff
@@ -76,6 +78,20 @@ fn generate_stocktake_lines(
         return generate_lines_initial_stocktake(connection, store_id, id);
     }
 
+    if let Some(true) = include_all_master_list_items {
+        let master_list_id = match master_list_id {
+            Some(id) => id,
+            None => {
+                return Err(RepositoryError::DBError {
+                    msg: "Master list ID is required when including all master list items"
+                        .to_string(),
+                    extra: "include_all_master_list_items is true".to_string(),
+                });
+            }
+        };
+        return generate_lines_from_master_list(connection, store_id, &id, &master_list_id);
+    }
```

Comment:
```suggestion
    if let Some(true) = include_all_master_list_items {
        match master_list_id {
            Some(master_list_id) => {
                return generate_lines_from_master_list(connection, store_id, id, &master_list_id)
            }
            None => {
                return Err(RepositoryError::DBError {
                    msg: "Master list ID is required when including all master list items"
                        .to_string(),
                    extra: "include_all_master_list_items is true".to_string(),
                });
            }
        };
    }
```

or just empty vec...